coordinator/src/main/scala/filodb.coordinator/queryplanner/SingleClusterPlanner.scala

```diff
@@ -387,18 +405,11 @@ class SingleClusterPlanner(val dataset: Dataset,
       assert(filters.size == 1, s"expected leaf plan to yield single filter group, but got ${filters.size}")
       leafPlan match {
         case rs: RawSeries =>
-          // Get time params from the RangeSelector, and use them to identify a TargetSchemaChanges.
           val startEndMsPairOpt = rs.rangeSelector match {
             case is: IntervalSelector => Some((is.from, is.to))
             case _ => None
           }
-          val tsLabels: Option[Seq[String]] = {
-            val tsFunc = tsp.targetSchemaFunc(filters.head)
-            startEndMsPairOpt match {
-              case Some((startMs, endMs)) => findTargetSchema(tsFunc, startMs, endMs).map(_.schema)
-              case _ => None
-            }
-          }
+          val tsLabels: Option[Seq[String]] = getTargetSchemaLabelsFromRawSeries(rs, qContext)
```

Comment:
Why a method? This lead to code duplication and doing ``map``  instead of pattern match will be concise as well